crates/engine/tree/src/tree/mod.rs

```diff
@@ -2180,14 +2181,25 @@ where
         let persistence_in_progress = self.persistence_state.in_progress();
         if !persistence_in_progress {
             state_root_result = match self
-                .compute_state_root_in_parallel(block.parent_hash, &hashed_state)
+                .compute_state_root_async(block.parent_hash, &hashed_state)
             {
                 Ok((state_root, trie_output)) => Some((state_root, trie_output)),
-                Err(ProviderError::ConsistentView(error)) => {
-                    debug!(target: "engine", %error, "Parallel state root computation failed consistency check, falling back");
+                Err(AsyncStateRootError::Provider(error)) => {
+                    debug!(target: "engine", %error, "Async state root computation failed provider error, falling back");
+                    None
+                }
+                Err(AsyncStateRootError::StorageRoot(error)) => {
+                    debug!(target: "engine", %error, "Async state root computation failed, falling back");
+                    None
+                }
+                Err(AsyncStateRootError::StorageRootChannelClosed { hashed_address }) => {
+                    debug!(target: "engine", %hashed_address, "Async state root computation failed, channel closed, falling back");
+                    None
+                }
+                Err(AsyncStateRootError::Receive(error)) => {
+                    debug!(target: "engine", %error, "Async state root computation failed, receive error, falling back");
```

Comment:
we can only ignore `AsyncStateRootError::Provider(ProviderError::ConsistentView(_))` here. every other error should be considered fatal

Reply:
@rkrasiuk ok, should we add a new variant to `InsertBlockErrorKindTwo` to consider the potential errors that come from the async calculation?

Reply:
done here https://github.com/paradigmxyz/reth/pull/10927/commits/ae131f9ac210d391804b2ab9cc1eb6dd10c3b91b